content/docs/tutorials/fullstack.md

```diff
@@ -35,20 +35,225 @@ In this tutorial, we will be going through the entire process of creating a full
+
+Initialize the needed files for `alembic` using the following command. We will need to edit some of the generated files to connect to our db.
+```bash
+alembic init src/alembic
+```
+Replace the code in your `src/alembic/env.py` with the following:
+{{<codesnippet `/source_code/diary_tutorial/alembic.py`>}}
+
+Now to make sure our values are registered in the `env.py` script, we need to set our environments. 
+
+```bash
+export SECRET_ARN=$(cdev output --value hello_world_comp.relationaldb.demo_db.secret_arn)
+```
+```bash
+export CLUSTER_ARN=$(cdev output --value hello_world_comp.relationaldb.demo_db.cluster_arn)
+```
+```bash
+export DB_NAME=<db_name>
+```
+{{<break 1>}}
+
+You should now be able to create automated migrations. 
+
+{{<tool_tip key="warning" summary="Auto Generation limits">}}
+You should familiarize yourself with the [limits of alembic auto generation](https://alembic.sqlalchemy.org/en/latest/autogenerate.html#what-does-autogenerate-detect-and-what-does-it-not-detect) and **ALWAYS** confirm the changes before applying them.
+{{</tool_tip>}}
+
+```bash
+alembic revision --autogenerate -m "Added entries table"
+```
+Apply the migration with the `upgrade` command. This will create our `Entry` Table.
+```bash
+alembic upgrade head
+```
+
+{{<break 1>}}
+Lets now connect to our DB and add an User.
+```bash
+cdev run relationaldb.shell hello_world_comp.demo_db
+```
+```sql
+BEGIN
+```
+```sql
+INSERT INTO entries(title, content) VALUES ('test entry','test our db connection');
+```
+```sql
+COMMIT
+```
+```bash
+quit
+```
+
+## Create Serverless Functions
+Now that the database is set up, it is time to add some serverless functions.  We can start by uncommenting lines 5-6, 13, 43-44, and 53-58 of your `src/hello_world/resources.py` file.
+{{<break >}}
+Then deploy the changes.
+```bash
+cdev deploy
+```
+
+Run the deployed function.
+```bash
+cdev run function.execute hello_world_comp.hello_world_function
+```
+
+Check the logs from the function
+```bash
+cdev run function.logs hello_world_comp.hello_world_function
+```
+Next, we will add the routes and functions to create and retrieve our diary entries. First, we will need to add a `src/hello_world/utils.py` file and add the following code to it.
+
+{{<codesnippet `/source_code/diary_tutorial/utils.py`>}}
+
+We will then add the following to our `src/hello_world/resources.py` file on line 14.
+```python
+from .utils import Response
+```
+
+Next, we need to add the following lines below the hello_route on line 24 of `src/hello_world/resources.py` so that our functions will have routes to be associated with.
+
+{{<codesnippet `/source_code/diary_tutorial/routes.py`>}}
+
+We can then add the function for creating entries below the hello function.
+
+{{<codesnippet `/source_code/diary_tutorial/create.py`>}}
+
+We can then add the entry serializer and function for retrieving all the entries below our create function.
+
+{{<codesnippet `/source_code/diary_tutorial/get.py`>}}
+
```

Comment:
Again in a similar vein as the above comment,  I worry about them having to hand edit these files as opposed to providing the final file and having them just paste it in.